src/test/java/com/uber/cadence/internal/testing/WorkflowTestingTest.java

```diff
@@ -729,4 +735,116 @@ public void testMockedChildSimulatedTimeout() {
+
+    @Override
+    public String workflow1(String input) {
+      // The test value should be in the MDC
+      return MDC.get("test");
+    }
+  }
+
+  @Test
+  public void testWorkflowContextPropagation() {
+    Worker worker =
+        testEnvironment.newWorker(
+            TASK_LIST,
+            builder ->
+                builder.setContextPropagators(
+                    Collections.singletonList(new TestContextPropagator())));
+    worker.registerWorkflowImplementationTypes(ContextPropagationWorkflowImpl.class);
+    testEnvironment.start();
+    MDC.put("test", "testing123");
+    WorkflowClient client = testEnvironment.newWorkflowClient();
+    WorkflowOptions options =
+        new WorkflowOptions.Builder()
+            .setContextPropagators(Collections.singletonList(new TestContextPropagator()))
+            .build();
+    TestWorkflow workflow = client.newWorkflowStub(TestWorkflow.class, options);
+    String result = workflow.workflow1("input1");
+    assertEquals("testing123", result);
+  }
+
+  public static class ContextPropagationParentWorkflowImpl implements ParentWorkflow {
+
+    @Override
+    public String workflow(String input) {
+      // Get the MDC value
+      String mdcValue = MDC.get("test");
+
+      // Fire up a child workflow
+      ChildWorkflowOptions options =
+          new ChildWorkflowOptions.Builder()
+              .setContextPropagators(Collections.singletonList(new TestContextPropagator()))
+              .build();
+      ChildWorkflow child = Workflow.newChildWorkflowStub(ChildWorkflow.class, options);
+
+      String result = child.workflow(mdcValue, Workflow.getWorkflowInfo().getWorkflowId());
+      return result;
+    }
+
+    @Override
+    public void signal(String value) {}
+  }
+
+  public static class ContextPropagationChildWorkflowImpl implements ChildWorkflow {
+
+    @Override
+    public String workflow(String input, String parentId) {
+      String mdcValue = MDC.get("test");
+      return input + mdcValue;
+    }
+  }
+
+  @Test
+  public void testChildWorkflowContextPropagation() {
+    Worker worker =
+        testEnvironment.newWorker(
+            TASK_LIST,
+            builder ->
+                builder.setContextPropagators(
+                    Collections.singletonList(new TestContextPropagator())));
+    worker.registerWorkflowImplementationTypes(
+        ContextPropagationParentWorkflowImpl.class, ContextPropagationChildWorkflowImpl.class);
+    testEnvironment.start();
+    MDC.put("test", "testing123");
+    WorkflowClient client = testEnvironment.newWorkflowClient();
+    WorkflowOptions options =
+        new WorkflowOptions.Builder()
+            .setContextPropagators(Collections.singletonList(new TestContextPropagator()))
+            .build();
+    ParentWorkflow workflow = client.newWorkflowStub(ParentWorkflow.class, options);
+    String result = workflow.workflow("input1");
+    assertEquals("testing123testing123", result);
+  }
```

Comment:
Add test to propagating down to an activity as well as test that tests that a context value set in a thread is propagated to a child thread.